src/routes.js

```
@@ -237,11 +247,16 @@ export default async function(
 
   router.get('/executions', async (req: ExpressRequest, res, next) => {
     try {
+
       const freeText = [];
-      if (req.query.freeText !== '') {
-        freeText.push(req.query.freeText);
+      if (typeof req.query.workflowId !== 'undefined' && req.query.workflowId.match(uuid_regex) !== null) {
+        freeText.push("(workflowId:" + req.query.workflowId  + ")");
       } else {
-        freeText.push('*');
+        freeText.push('(*)');
+      }
+
+      if (typeof req.query.status !== 'undefined' &&  WORKFLOW_STATUS_TYPES.includes(req.query.status)) {
+        freeText.push("(status:" + req.query.status  + ")");
       }
```

What if user submits workflow status type with a type-o ? it is silently ignored, instead we should return an error

same for req.query.workflowId -> return error if it isn't matching regex